rust/worker/Dockerfile.compactor

```diff
@@ -0,0 +1,40 @@
+FROM rust:1.74.1 as builder
```

Comment:
Yes please! That way they can share the first 23 lines of this file and cache them together.

Starting at line 23, where we create the dummy `main`s and build the binaries, we need separate builders so that changes to one only affect the other. We want to do exactly what we do now: the only differences in the below are the new `FROM builder as compactor-builder` and the name of the final image (`... as compactor`).

```
FROM builder as compactor-builder
# We need to replace the compaction node's real main() with a dummy at the expected location.
RUN mkdir -p rust/worker/src/bin/ && echo "fn main() {}" > rust/worker/src/bin/compaction_service.rs
RUN if [ "$RELEASE_MODE" = "1" ]; then cargo build --bin compaction_service --release; else cargo build --bin compaction_service; fi
RUN rm -rf rust/

COPY rust/ rust/
RUN touch rust/worker/src/bin/compaction_service.rs
RUN if [ "$RELEASE_MODE" = "1" ]; then cargo build --bin compaction_service --release; else cargo build --bin compaction_service; fi
RUN if [ "$RELEASE_MODE" = "1" ]; then mv target/release/compaction_service .; else mv target/debug/compaction_service .; fi

FROM debian:bookworm-slim as compactor

...
```

And then the same for the query service.